src/api/result_proxy_api_client.rs

```diff
@@ -0,0 +1,324 @@
+    ///     }
+    /// }
+    /// ```
+    pub fn upload_to_ipfs(
+        &self,
+        authorization: &str,
+        result_model: &ResultModel,
+    ) -> Result<String, reqwest::Error> {
+        let url = format!("{}/v1/results", self.base_url);
+        let response = self
+            .client
+            .post(&url)
+            .header("Authorization", authorization)
+            .json(result_model)
+            .send()?;
+
+        if response.status().is_success() {
+            response.text()
+        } else {
+            Err(response.error_for_status().unwrap_err())
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::json;
+    use wiremock::{
+        Mock, MockServer, ResponseTemplate,
+        matchers::{body_json, header, method, path},
+    };
+
+    // Test constants
+    const TEST_TASK_ID: &str = "0x123";
+    const TEST_DEAL_ID: &str = "0x456";
+    const TEST_DETERMINISTIC_HASH: &str = "0xabc";
+    const TEST_ENCLAVE_SIGNATURE: &str = "0xdef";
+    const TEST_IPFS_LINK: &str = "ipfs://QmHash123";
+    const TEST_TOKEN: &str = "test-token";
+
+    // region ResultModel
+    #[test]
+    fn result_model_default_returns_correct_values_when_created() {
+        let model = ResultModel::default();
+        assert_eq!(model.chain_task_id, EMPTY_HEX_STRING_32);
+        assert_eq!(model.deal_id, EMPTY_HEX_STRING_32);
+        assert_eq!(model.task_index, 0);
+        assert!(model.zip.is_empty());
+        assert_eq!(model.deterministic_hash, "");
+        assert_eq!(model.enclave_signature, EMPTY_WEB3_SIG);
+    }
+
+    #[test]
+    fn result_model_serializes_to_camel_case_when_converted_to_json() {
+        let model = ResultModel {
+            chain_task_id: String::from(TEST_TASK_ID),
+            deal_id: String::from(TEST_DEAL_ID),
+            task_index: 5,
+            zip: vec![1, 2, 3],
+            deterministic_hash: String::from(TEST_DETERMINISTIC_HASH),
+            enclave_signature: String::from(TEST_ENCLAVE_SIGNATURE),
+        };
+
+        let expected = json!({
+            "chainTaskId": TEST_TASK_ID,
+            "dealId": TEST_DEAL_ID,
+            "taskIndex": 5,
+            "zip": [1, 2, 3],
+            "deterministicHash": TEST_DETERMINISTIC_HASH,
+            "enclaveSignature": TEST_ENCLAVE_SIGNATURE
+        });
+
+        let v = serde_json::to_value(model).unwrap();
+        assert_eq!(v, expected);
+    }
+
+    #[test]
+    fn result_model_deserializes_from_camel_case_when_parsing_json() {
+        let json_str = format!(
+            r#"{{
+            "chainTaskId": "{}",
+            "dealId": "{}",
+            "taskIndex": 5,
+            "zip": [1, 2, 3],
+            "deterministicHash": "{}",
+            "enclaveSignature": "{}"
+        }}"#,
+            TEST_TASK_ID, TEST_DEAL_ID, TEST_DETERMINISTIC_HASH, TEST_ENCLAVE_SIGNATURE
+        );
+
+        let model: ResultModel = serde_json::from_str(&json_str).unwrap();
```

Comment:
Is it really intended to use `from_str` here or could `from_value` have been used with the `json` macro to create the JSON data ?